torch/_dynamo/backends/distributed.py

```diff
@@ -509,18 +518,24 @@ def compile_fn(self, gm: fx.GraphModule, example_inputs: List[torch.Tensor]):
                     if buckets[0].opcount_increased_to_capture_external_output == 0:
                         buckets[0].paramsize_before_opcount_increase = buckets[0].size
                     buckets[0].opcount_increased_to_capture_external_output += 1
-
             if node.op == "call_module":
-                target = gm.get_submodule(node.target)
-                for name, param in target.named_parameters():
-                    if param.requires_grad and not self._ignore_parameter(param):
-                        buckets[0].size += param.untyped_storage().nbytes()
-                        buckets[0].params.append(f"{node.target}_{name}")
-                        buckets[0].param_ids.append(id(param))
+                target_mod = gm.get_submodule(node.target)
+                if target_mod not in processed_modules:
+                    self.add_module_params_to_bucket(target_mod, buckets[0], processed_modules, node.target)
+            elif node.op == "call_method":
+                try:
+                    target_mod = gm.get_submodule(node.args[0].target)
+                    if isinstance(target_mod, torch.nn.Module) and target_mod not in processed_modules:
+                        self.add_module_params_to_bucket(target_mod, buckets[0], processed_modules, node.target)
+                except AttributeError:
```

Comment:
Instead of doing `try ... except ...`, is there another way to detect the `call_method with no target module` case and not do anything in that case?